src/Microsoft.Health.Dicom.SqlServer/Features/Workitem/SqlWorkitemStore.cs

```diff
@@ -26,35 +26,39 @@ public SqlWorkitemStore(VersionedCache<ISqlWorkitemStore> cache)
         public async Task<long> BeginAddWorkitemAsync(int partitionKey, DicomDataset dataset, IEnumerable<QueryTag> queryTags, CancellationToken cancellationToken = default)
         {
             ISqlWorkitemStore store = await _cache.GetAsync(cancellationToken: cancellationToken);
+
             return await store.BeginAddWorkitemAsync(partitionKey, dataset, queryTags, cancellationToken);
         }
 
-        public async Task BeginUpdateWorkitemAsync(WorkitemMetadataStoreEntry workitemMetadata, CancellationToken cancellationToken = default)
+        public async Task<(long WorkitemKey, long Watermark)?> BeginAddWorkitemWithWatermarkAsync(int partitionKey, DicomDataset dataset, IEnumerable<QueryTag> queryTags, CancellationToken cancellationToken = default)
```

Comment:
Why adding new methods? Why cant we overide the existing method?

Reply:
Instead of doing this, we should return VersionedWorkitemInstanceidentifier similar to STORE. Then we can avoid having 2 methods